### DISCUSSION TITLE
How to keep some NFTs for yourself?

### DISCUSSION THREAD

**Discussion by fullStackRyan** [hashlips-lab/nft-erc721-collection#234] [Q&A]
Hi, 

1. I want to hold back 10 or so NFTs for my team. Is it possible to do that and if so, how?
2. Is it also possible to specify specific NFTs you want to keep?

Thanks :)

**Comment by ecerroni** [comment]
The easiest way would be to mint the first 10 tokens for your team at contract deploy. When you create your random NFTs you will know that the first 10 NFT are not actually random and will be minted at contract deploy with an onlyOwner function.

Then you transfer them from contract owner to your team members addressess.

There are more sophisticated ways to achieve similar results, but they would require more effort.

**Reply by fullStackRyan** [reply]
So would I set the mint price to 0. Then mint the first 10? I guess I would just pay minimal gas fees as the price is 0?

**Reply by ecerroni** [reply]
You may use this code and either invoke it in the constructor or after deploy

```
  function ownerMint(uint256 _mintAmount) public onlyOwner {
    uint256 supply = totalSupply();
    require(supply + _mintAmount <= maxSupply, "Max supply exceeded!");

    if (_mintAmount > 0) {
      _safeMint(_msgSender(), _mintAmount);
    }
  }
```

**Reply by Hofstud** [reply]
Thank you for this cool snippet. How would you code if the collection is 10,000 and you want to mint the 1st 10% for the creators and team (1,000 NFT)?

**Comment by m360ai** [comment]
Depends how big your launch is and how its marketed. If you deploy it
before the "public" mint date, you can whitelist your own wallet(s) set the
price at what you want which could be as low or high as desired for
yourself and your team, then mint with the whitelisted wallet(s) prior to
public launch, then reset the price to what you want for the public.
Specific NFTs hmm, not through the mint process that I'm aware of. Marco,
would know more on this.

**Reply by Hofstud** [reply]
Thanks for confirming as this is how I was going to do the initial minting but would prefer to have to code automatically mint to owner

**Comment by NFTmetaman** [comment]
I believe it would look something like this (to mint specific id's)

uint256 public constant MAX_TOKEN_ID = 10000;

_mint(msg.sender, id);

**Comment by it-is-zods** [comment]
as @ecerroni mentioned, the easiest way is to do mint the first ones.
you can use mintForAddress() already in the contract. It is only available to the owners and can be done even if the contract is paused
the addd bonus is that you can send it to any wallet as the arguments are _mintAmount and _receiver
also, you only pay for gas
